File: src-tauri/src/models/guest.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Clone, Debug, sqlx::FromRow)]
pub struct Guest {
    pub id: i64,
    pub business_id: i32,
    pub name: String,
    pub phone: String,
    pub verify_id: i32,
    pub status_id: i32,
    pub created_at: Option<String>,
    pub updated_at: Option<String>,
}

#[derive(Serialize, Deserialize, Clone, Debug)]
pub struct GuestCreate {
    pub business_id: i32,
    pub name: String,
    pub phone: String,
    pub verify_id: i32,
    pub status_id: i32,
}

#[derive(Serialize, Deserialize, Clone, Debug)]
pub struct GuestUpdate {
    pub name: Option<String>,
    pub phone: Option<String>,
    pub verify_id: Option<i32>,
    pub status_id: Option<i32>,
}

File: src/Types/guest.ts
export interface Guest {
    id: number;
    business_id: number;
    name: string;
    phone: string;
    verify_id: number;
    verify_name: string;
    status_id: number;
    status_name: string;
    created_at: string;
    updated_at: string;
}

export interface GuestResponse{
    data: Guest[];
}